unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentView.cs

```diff
@@ -271,21 +319,121 @@ public void SetCollectibleEmotes(WearableItem[] wearables)
+                            showType = true,
+                            type = "name",
+                            marketplaceURI = "",
+                            name = names[i + j].Name,
+                            rarity = "legendary",
+                            imageURI = ""
+                        };
+
+                        nftIds[j] = (names[i + j].ContractAddress, "name");
+                    }
+                    else
+                    {
+                        pageElements[j] = null;
+                        nftIds[j] = ("", "");
+                    }
+                }
+                nftPageView.SetPageElementsContent(pageElements, nftIds);
+                nftPageView.IsNftInfoActived = false;
+                nftPageView.OnClickBuyNft += ClickOnBuyWearable;
+                nftPageView.OnFocusAnyNtf += FocusAnyNtfItem;
+                nftNamesCarousel.AddItem(nftPageView);
+            }
+            nftNamesCarousel.GenerateDotsSelector();
+        }
+
+        public void SetCollectibleLands(LandsResponse.LandEntry[] lands)
+        {
+            nftLandsCarousel.gameObject.SetActive(lands.Length > 0);
+            nftLandsCarousel.CleanInstantiatedItems();
+            emptyLandsText.SetActive(lands.Length <= 0);
+            for (int i = 0; i < lands.Length; i += 4)
+            {
+                PoolableObject nftPagePoolElement = nftPagesEntryPool.Get();
+                nftPagesPoolableQueue.Add(nftPagePoolElement);
+                nftPagePoolElement.gameObject.transform.SetParent(nftLandsCarouselContent, false);
+                NftPageView nftPageView = nftPagePoolElement.gameObject.GetComponent<NftPageView>();
+                nftPageView.OnClickBuyNft -= ClickOnBuyWearable;
+                nftPageView.OnFocusAnyNtf -= FocusAnyNtfItem;
+                NFTIconComponentModel[] pageElements = new NFTIconComponentModel[4];
+                (string, string)[] nftIds = new (string, string)[4];
+                for (int j = 0; j < 4; j++)
+                {
+                    if (lands.Length > i + j && lands[i + j] != null)
+                    {
+                        pageElements[j] = new()
+                        {
+                            showMarketplaceButton = true,
+                            showType = true,
+                            type = lands[i + j].Category,
+                            marketplaceURI = "",
+                            name = !string.IsNullOrEmpty(lands[i + j].Name) ? lands[i + j].Name : lands[i + j].Category,
+                            rarity = "land",
+                            imageURI = lands[i + j].Image
+                        };
+
+                        nftIds[j] = (lands[i + j].ContractAddress, lands[i + j].Category);
+                    }
+                    else
+                    {
+                        pageElements[j] = null;
+                        nftIds[j] = ("", "");
+                    }
+                }
```

Comment:
same here. I would suggest to put it inside the method with a good name, that describe in high-level words what happening here

Reply:
done